File: src/solutions/002/002_arvindch.py
#!/usr/bin/env python2.7

__author__ = 'Arvind Chembarpu'
__email__ = 'achembarpu@gmail.com'


from src.custom import gens


def main():
    
    giv_lim = 4000000
    req_sum = 0
    
    for fib_num in gens.fibonacci_gen():
        if fib_num > giv_lim:
            break
        if fib_num % 2 == 0:
            req_sum += fib_num
    
    return req_sum


if __name__ == '__main__':
    answer = main()
    print answer

File: src/java/solutions/002/_002_arvindch.java
public class _002_arvindch {

	public static void main(String[] args) {
		int a=0, b=1, c=1, giv_lim=4000000, req_sum=0;
		
		while(c < giv_lim) {
			
			if(c%2==0) {
				req_sum += c;
			}
			
			c = a + b;
			a = b;
			b = c;
		}
		
		System.out.println(req_sum);
	}

}
